### ISSUE TITLE
Table footer no longer fixed with when scrollable

### ISSUE THREAD

**Issue by monsieur-ricky** [primefaces/primeng#17224]
### Describe the bug

While migrating my app from PrimeNG v17 to version 19, I noticed that the table component is no longer working as expected.
The main issue that I'm reporting is the footer is no longer fixed when the table is scrollable. This issue can be tested in the official documents:

Image from the version 17 docs:
<img width="940" alt="Screenshot 2024-12-26 at 11 59 47" src="https://github.com/user-attachments/assets/6f4a437b-72cc-4166-be7e-9c2592416f65" />

Image from the version 19 docs:
<img width="940" alt="Screenshot 2024-12-26 at 11 58 55" src="https://github.com/user-attachments/assets/e605fcab-f338-424d-973e-542121d39e27" />

This is quite annoying because the table footer has relevant summary information that now requires the user to do a long scroll to the bottom of the table to view that information.

### Reason for not contributing a PR

- [ ] Lack of time
- [X] Unsure how to implement the fix/feature
- [ ] Difficulty understanding the codebase
- [ ] Other

### Reproducer

https://stackblitz.com/edit/baymtu58-2shz8rsa

### Environment

Angular CLI: 19.0.6
Node: 20.16.0
Package Manager: npm 10.9.2
OS: darwin x64

### Angular version

19.0.6

### PrimeNG version

v19

### Node version

20.16.0

### Browser(s)

Chrome 131.0.6778.205

### Steps to reproduce the behavior

To reproduce the issue:
1. Access the official Table component documentation for version 19
2. Go to the horizontal scroll demo: https://primeng.org/table#horizontal-scroll
3. Scroll the table to check that the footer **isn't fixed**.

### Expected behavior

To check the correct and desired behaviour:
1. Access the official Table component documentation for version 17
2. Go to the horizontal scroll demo: https://v17.primeng.org/table#horizontal-scroll
3. Scroll the table to confirm that the footer **is fixed**.

**Comment by vcucchiara** [comment]
I have this same issue in v18. As a workaround, I added this to my component css

:host ::ng-deep .p-datatable.p-datatable-sm .p-datatable-tfoot {
  position:sticky;
}

**Comment by monsieur-ricky** [comment]
Yes, I've also did a temporary fix using CSS. Hopefully this will be fixed in 19.0.3.
